net/server.go

```diff
@@ -51,11 +51,6 @@ type server struct {
 	pubSubEmitter  event.Emitter
 	pushLogEmitter event.Emitter
 
```

Comment:
suggestion: If you remove the `docQueue`, it opens the door for transaction conflicts on the dag sync process. One of the test matrix run failed for this exact reason. I suggest you keep it in :)

Reply:
It's still there but in the `db` package instead. I think it makes more sense to have the blocking occur there.

Reply:
The problem is that if the conflict happens in exactly the wrong way, blocks will be missing from the blockstore. I wanted to remove it when I refactored the dag sync process and I found out that we have to keep it in. Alternatively we need to change the dags sync logic to deal with conflicts.

Reply:
Oh right I remember now. Would a simple fix be to ignore transaction conflict errors?

Reply:
`GetBlocks` doesn't handle errors very well. Even `GetBlock` doesn't return the block on `put` conflict so we would need to retry. It would be an efficien retry because the block would be found locally on the next try. I'll let you decide what to do from here :)

Reply:
The updated sync process should now handle conflicts.